tests/src/FunctionalJavascript/CorporateContactFormTest.php

```diff
@@ -0,0 +1,320 @@
+    /** @var \Behat\Mink\WebAssert $assert */
+    $assert = $this->assertSession();
+    /** @var \Behat\Mink\Element\DocumentElement $page */
+    $page = $this->getSession()->getPage();
+
+    $this->drupalGet('admin/structure/contact/add');
+
+    // Assert corporate fields are not present before is_corporate_form click.
+    $is_corporate_form = $page->findField('is_corporate_form');
+    $this->assertNotEmpty($is_corporate_form);
+    $this->assertFieldsAreMissing();
+
+    // Ajax call to load corporate fields.
+    $is_corporate_form->click();
+    $assert->assertWaitOnAjaxRequest();
+
+    // Assert fields are now visible.
+    $this->assertFieldsAreVisible();
+
+    // Assert expose_as_block is checked by default.
+    $element = $page->findField('corporate_fields[expose_as_block]');
+    $this->assertNotEmpty($element);
+    $this->assertTrue($element->isChecked());
+
+    // Add contact required values.
+    $this->fillCoreContactFields();
+
+    // Test topic ajax.
+    $this->assertTopicAddAjax();
+
+    // Test remove topic group.
+    $this->assertTopicRemoveAjax();
+
+    // Add remaining field values and submit form.
+    $this->fillCorporateFields();
+    $page->pressButton('Save');
+    $assert->pageTextContains('Contact form Test form has been added.');
+
+    // Assert the values are saved.
+    $this->drupalGet('admin/structure/contact/manage/oe_corporate_form');
+    $is_corporate_form = $page->findField('is_corporate_form');
+    $this->assertNotEmpty($is_corporate_form);
+    $this->assertTrue($is_corporate_form->isChecked());
+
+    // Make sure the saved values are the ones expected.
+    $this->checkCorporateFieldsOnPage();
+    $this->checkCorporateFieldsInStorage();
+  }
+
+  /**
+   * Tests the corporate contact form.
+   */
+  public function testNoCorporateValues(): void {
+    /** @var \Behat\Mink\WebAssert $assert */
+    $assert = $this->assertSession();
+    /** @var \Behat\Mink\Element\DocumentElement $page */
+    $page = $this->getSession()->getPage();
+
+    $this->drupalGet('admin/structure/contact/add');
+
+    // Assert corporate fields are not present.
+    $is_corporate_form = $page->findField('is_corporate_form');
+    $this->assertNotEmpty($is_corporate_form);
+    $this->assertFieldsAreMissing($page);
+
+    // Ajax call to load corporate fields.
+    $is_corporate_form->click();
+    $assert->assertWaitOnAjaxRequest();
+
+    // Click again to remove them.
+    $is_corporate_form->click();
+    $assert->assertWaitOnAjaxRequest();
+
+    // Check nothing changed.
+    $this->assertFieldsAreMissing($page);
+
+    // Add contact required values.
+    $this->fillCoreContactFields($page);
+
+    $page->pressButton('Save');
+    $assert->pageTextContains('Contact form Test form has been added.');
+
+    // Assert the values are saved.
+    $this->drupalGet('admin/structure/contact/manage/oe_corporate_form');
+    $this->assertFieldsAreMissing($page);
```

Comment:
We could also assert that third party settings are clean no?

Reply:
done